pkg/session/session.go

```diff
@@ -208,15 +213,56 @@ func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *
 	return c, nil
 }
 
-func clearCache(sessionKey string) {
-	sessionMU.Lock()
-	defer sessionMU.Unlock()
-	delete(sessionCache, sessionKey)
+func clearCache(logger logr.Logger, sessionKey string) {
+	if cachedSession, ok := sessionCache.Load(sessionKey); ok {
+		s := cachedSession.(*Session)
+
+		// we check for tagmanager session
+		// because calling on expired session blocks
+		session, err := s.TagManager.Session(context.Background())
+		if err != nil {
+			logger.Error(err, "unable to get tag manager session")
+		}
+		if session != nil {
+			logger.Info("found active tag manager session, logging out")
+			err := s.TagManager.Logout(context.Background())
+			if err != nil {
+				logger.Error(err, "unable to logout tag manager session")
+			}
+		}
+
+		vimSessionActive, err := s.SessionManager.SessionIsActive(context.Background())
+		if err != nil {
+			logger.Error(err, "unable to get vim client session")
+		} else if vimSessionActive {
+			logger.Info("found active vim session, logging out")
+			err := s.SessionManager.Logout(context.Background())
+			if err != nil {
+				logger.Error(err, "unable to logout vim session")
+			}
+		}
+	}
+	sessionCache.Delete(sessionKey)
 }
 
 // newManager creates a Manager that encompasses the REST Client for the VSphere tagging API.
-func newManager(ctx context.Context, client *vim25.Client, user *url.Userinfo) (*tags.Manager, error) {
+func newManager(ctx context.Context, logger logr.Logger, sessionKey string, client *vim25.Client, user *url.Userinfo, feature Feature) (*tags.Manager, error) {
 	rc := rest.NewClient(client)
+	if feature.EnableKeepAlive {
+		rc.Transport = keepalive.NewHandlerREST(rc, feature.KeepAliveDuration, func() error {
+			s, err := rc.Session(ctx)
+			if err != nil {
+				return err
+			}
+			if s != nil {
+				return nil
+			}
+
+			logger.Info("rest client session expired, clearing cache")
```

Comment:
```suggestion
			logger.V(6).Info("rest client session expired, clearing cache")
```
to make logs less noisy

Reply:
Ack